### ISSUE TITLE
Lazy load and AOT : Working but splitted chunk ONLY WITH AOT FLAG

### ISSUE THREAD

**Issue by maxime1992** [angular/angular-cli#3267]
> Please provide us with the following information:
> ---------------------------------------------------------------

### OS?
Ubuntu 16.10 (x64)

### Versions.
angular-cli: 1.0.0-beta.21

### Repro steps.
After I read more about lazy load + AOT on this [issue](https://github.com/angular/angular-cli/issues/2452), I decided to give a try and it's working really nice for the first time in my app.

So I decided to have some fun and benchmark different results. Here's the ouput : 
```
+-----------------------+-------------+--------------+-----------+-------------+
|                       | Main bundle |   Chunk 0    | Scripting | First paint |
+-----------------------+-------------+--------------+-----------+-------------+
| ng serve              | 4.5 MB      | Not splitted | 6075 ms   | 5500+ ms    |
| ng serve --prod       | 334 KB      | Not splitted | 5587 ms   | 4750+ ms    |
| ng serve --aot        | 3.3 MB      | 326 KB       | 4011 ms   | 4400+ ms    |
| ng serve --prod --aot | 243 KB      | 18.1 Kb      | 3860 ms   | 4250+ ms    |
+-----------------------+-------------+--------------+-----------+-------------+
```
- The --prod --aot build size is 27% smaller than --prod build 
- The --prod --aot build is 31% faster when scripting than --prod build 
- AOT is cool ! 

**BUT**

As you can see, `ng serve` and `ng serve --prod` doesn't have the chunk expected and I'm able to find the code (that's supposed to be in the chunk) inside the main bundle. With `aot` flag, I always have both the main bundle and my chunk.

Even tho it's working, devs who want to put their app in prod without building with `--aot` will not enjoy the lazy loading they've setup ;)

**Comment by MrCroft** [comment]
It works fine on my end (here: https://github.com/MrCroft/lazy)
Plus I already had a bigger app and updated the cli from beta.20-4 to beta.21, and `ng build --prod` or `ng serve --prod` both work as before (separate chunks get created for each lazy module).
Make sure you're using angular 2.2.1 (they've locked the version to that I see, in the latest CLI release).
Ultimately, if the app wasn't initially generated with beta.21, remove node_modules and do an `ng init`, check the file differences.

**Comment by intellix** [comment]
Also noticed this. I have a 3.2mb main bundle only now

**Comment by aaronleeucla** [comment]
AOT has never worked for me, due to at least two open issues elsewhere.

I upgraded from Angular 2.1.2 to 2.2.1 and CLI from beta 18 to beta 21. Instantly lost lazy-loading; no longer seeing any chunks but one big main bundle.

**Comment by filipesilva** [comment]
That's definitely odd. You should always have the lazy loaded chunks. 

Does this still happen with the latest versions? If so, can you get me a repro I can debug?

**Comment by intellix** [comment]
Not happening for me anymore with the latest stuff (b24)

**Comment by maxime1992** [comment]
I've got many errors since beta 24 which make me stay on beta 22.
(for example : `No module factory available for dependency type: ContextElementDependency`).
When I manage to make my project works with beta 24 I'll let you know about the lazy loading problem. :)

**Comment by filipesilva** [comment]
Dupe of #2496 (I think), fixed in #3079.

**Comment by shawnlan** [comment]
I'm running angular cli beta 26. Testing on an empty project (`ng new`).

- built with aot: got 0.bundle.js.
- built without aot: got no chunk file.

I know this is an empty project, which doesn't have any lazy module. But is this issue really fixed?

And just curious. What special thing does AOT do to make Webpack code splitting? I checked the `tsconfig.json`. Typescript modules are compiled into `es6` modules (`import` and `export` keywords), and Webpack recognizes `es6` modules, right? Isn't it enough for Webpack to do code splitting? What else is needed?
